suite-native/transactions/src/components/TransactionsList/TransactionList.tsx

```diff
@@ -1,37 +1,62 @@
 import React from 'react';
-import { ActivityIndicator } from 'react-native';
+import { ActivityIndicator, SectionList } from 'react-native';
 
 import { A } from '@mobily/ts-belt';
 
 import { useNativeStyles } from '@trezor/styles';
-import { WalletAccountTransaction } from '@suite-common/wallet-types';
+import { Account, WalletAccountTransaction } from '@suite-common/wallet-types';
 import { groupTransactionsByDate } from '@suite-common/wallet-utils';
+import { Box, Text } from '@suite-native/atoms';
+import { AccountBalance } from '@suite-native/accounts';
 
+import { TransactionListGroupTitle } from './TransactionListGroupTitle';
 import { TransactionListItem } from './TransactionListItem';
-import { TransactionListGroup } from './TransactionListGroup';
 
 type AccountTransactionProps = {
     transactions: WalletAccountTransaction[];
+    accountName?: string;
+    account: Account;
+    fetchMoreTransactions: () => void;
 };
 
-export const TransactionList = ({ transactions }: AccountTransactionProps) => {
+export const TransactionList = ({
+    transactions,
+    accountName,
+    account,
+    fetchMoreTransactions,
+}: AccountTransactionProps) => {
     const { utils } = useNativeStyles();
     const accountTransactionsByDate = groupTransactionsByDate(transactions);
     const transactionDateKeys = Object.keys(accountTransactionsByDate);
 
+    const formatSectionData = () => {
+        const sectionsData: { dateKey: string; data: WalletAccountTransaction[] }[] = [];
+        transactionDateKeys.forEach(dateKey => {
+            sectionsData.push({ dateKey, data: [...accountTransactionsByDate[dateKey]] });
+        });
+        return sectionsData;
+    };
+
     if (A.isEmpty(transactionDateKeys))
         // TODO Temporary loading state just so it's visible that transactions are loading
         return <ActivityIndicator size="large" color={utils.colors.forest} />;
 
     return (
-        <>
-            {transactionDateKeys.map(dateKey => (
-                <TransactionListGroup dateKey={dateKey} key={dateKey}>
-                    {accountTransactionsByDate[dateKey].map(transaction => (
-                        <TransactionListItem key={transaction.txid} transaction={transaction} />
-                    ))}
-                </TransactionListGroup>
-            ))}
-        </>
+        <SectionList
+            sections={formatSectionData()}
+            renderSectionHeader={({ section: { dateKey } }) => (
+                <TransactionListGroupTitle key={dateKey} dateKey={dateKey} />
+            )}
+            renderItem={({ item }) => <TransactionListItem key={item.txid} transaction={item} />}
```

Comment:
This should be wrapped in `useCallback` also whole `TransactionListItem` itself should use `React.memo`

These are like mandatory things that must be always used for FlatList/SectionList in RN

Reply:
[f3a77d8](https://github.com/trezor/trezor-suite/pull/6319/commits/f3a77d8c2b615620ef6ce5d2c88de78da97ac147)